python/triton/language/core.py

```diff
@@ -408,7 +408,7 @@ def __init__(self, handle, type: dtype):
         # IR handle
         self.handle = handle
         # Block shape
-        self.shape = (1, )
+        self.shape = ()
```

Comment:
The original shape is actually wrong: it returns `(1, )` even when it's a scalar. We didn't find this error because many operators infer their output shapes from input shapes and do not use the shapes here. However, `ExtElemwiseOp` uses this shape and the ret type is wrong before this modification

Reply:
There's a fix to this problem in master, but I just haven't ported it to `triton-mlir` yet. I'll do it sometime today.

Reply:
https://github.com/openai/triton/pull/839

Can you try this out and see if it fixes your problem? Using `tl.libdevice.exp` but not `tl.exp`.

Reply:
I just checked and can confirm that the changes in #839 fixes this problem successfully @Jokeren

Reply:
Got you, I'll merge #839 into triton-mlir to facilitate your PR.